client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/IndexedCacheNonIndexedEntityTest.java

```diff
@@ -0,0 +1,112 @@
+import org.infinispan.protostream.annotations.ProtoSchemaBuilder;
+import org.infinispan.query.remote.client.ProtobufMetadataManagerConstants;
+import org.infinispan.server.core.admin.embeddedserver.EmbeddedServerAdminOperationHandler;
+import org.infinispan.server.hotrod.HotRodServer;
+import org.infinispan.server.hotrod.configuration.HotRodServerConfigurationBuilder;
+import org.infinispan.test.SingleCacheManagerTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+/**
+ * @since 12.0
+ */
+@Test(testName = "client.hotrod.query.IndexedCacheNonIndexedEntityTest", groups = "functional")
+public class IndexedCacheNonIndexedEntityTest extends SingleCacheManagerTest {
+   private static final String CACHE_NAME = "IndexedCacheNonIndexedEntitiesTest";
+
+   protected HotRodServer hotRodServer;
+   protected RemoteCacheManager remoteCacheManager;
+
+   @ProtoName("Entity")
+   static class Entity {
+      private final String name;
+
+      @ProtoFactory
+      public Entity(String name) {
+         this.name = name;
+      }
+
+      @ProtoField(number = 1)
+      public String getName() {
+         return name;
+      }
+   }
+
+   private String createProtoFile() throws IOException {
+      SerializationContext serializationContext = MarshallerUtil.getSerializationContext(remoteCacheManager);
+      ProtoSchemaBuilder protoSchemaBuilder = new ProtoSchemaBuilder();
+      return protoSchemaBuilder.fileName("file.proto")
+            .addClass(Entity.class)
+            .build(serializationContext);
+   }
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager() throws Exception {
+      cacheManager = TestCacheManagerFactory.createCacheManager(new ConfigurationBuilder());
+
+      HotRodServerConfigurationBuilder serverBuilder = new HotRodServerConfigurationBuilder();
+      serverBuilder.adminOperationsHandler(new EmbeddedServerAdminOperationHandler());
+      hotRodServer = HotRodClientTestingUtil.startHotRodServer(cacheManager, serverBuilder);
+
+      org.infinispan.client.hotrod.configuration.ConfigurationBuilder clientBuilder = newRemoteConfigurationBuilder();
+      clientBuilder.addServer().host("127.0.0.1").port(hotRodServer.getPort());
+      remoteCacheManager = new RemoteCacheManager(clientBuilder.build());
+
+      RemoteCache<String, String> metadataCache = remoteCacheManager.getCache(PROTOBUF_METADATA_CACHE_NAME);
+      metadataCache.put("file.proto", createProtoFile());
+      assertFalse(metadataCache.containsKey(ProtobufMetadataManagerConstants.ERRORS_KEY_SUFFIX));
+
+      return cacheManager;
+   }
+
+   @AfterClass(alwaysRun = true)
+   public void release() {
+      killRemoteCacheManager(remoteCacheManager);
+      killServers(hotRodServer);
+   }
+
+   @Test(expectedExceptions = HotRodClientException.class, expectedExceptionsMessageRegExp = ".*The configured indexed-entity type 'Entity' must be indexed.*")
```

Comment:
instead of using expected exception, we should be using the Exceptions utility and assert in the test

Reply:
I looked at that utility, but I prefer this style, it's more concise and expressive.

Reply:
The directive I got in the past is to use the utility instead of expected exceptions,  like where we wanted to move on our code base tests. @danberindei any thoughts ?

Reply:
Move from ```testng``` -> ```junit``` you mean? In this case, I prefer to use junit specifc helpers rather than the utility
https://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html